### PR TITLE
Fetch test

### PR DESCRIPTION
**Pull request by ashwinikamble-sacumen**
for the fetch function test_fetch created

### CODE REVIEW

File: src/fetch.py [hunk 1]
```diff
@@ -7,8 +7,8 @@ def fetch(url):
         response = requests.get(url)
         if response.status_code == 200:
             return response.json()
-        else:
-            raise RuntimeError(f"Request failed {response.status_code}")
+        
+        raise RuntimeError(f"Request failed {response.status_code}")
     except requests.exceptions.RequestException as e:
         """log error message"""
```
Comment (vinodkumar-sacumen) [inline src/fetch.py:13]: Why do you need raise exception when you already have try catch block?

File: tests/test_fetch.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from src.fetch import fetch
+
+
+def test_fetch(caplog):
+    url = "https://api.publicapis.org/entries"
+    caplog.set_level(logging.ERROR)
+    result = fetch(url)
+    assert result is not None
+    assert len(caplog.records)==0
+    caplog.handler.stream.close()
```
Comment (vinodkumar-sacumen) [inline tests/test_fetch.py:14]: Test case is wrong.

**Review by vinodkumar-sacumen — CHANGES_REQUESTED** [review]

File: src/fetch.py (hunk 1 above)
Comment (ashwinikamble-sacumen) [reply src/fetch.py:13]: yes there is no need  will correct it

File: tests/test_fetch.py (hunk 2 above)
Comment (ashwinikamble-sacumen) [reply tests/test_fetch.py:14]: ok

File: tests/test_fetch.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    mocker.patch("requests.get", mock_api_failure)
+    response = fetch(url="test_url")
+    assert response == "wrong url"
+
+
+@pytest.mark.vcr
+def test_fetch_with_vcr():
+    response = fetch(url="https://api.publicapis.org/entries")
+    assert response == 1425
+
```
Comment (vinodkumar-sacumen) [inline tests/test_fetch.py:33]: I doubt these test cases will work.

**Review by vinodkumar-sacumen — CHANGES_REQUESTED** [review]

File: tests/test_fetch.py (hunk 3 above)
Comment (ashwinikamble-sacumen) [reply tests/test_fetch.py:33]: I check out these test case are passed

**Pull request merged**